speed.py

```diff
@@ -0,0 +1,174 @@
+    """
+    Creates and returns a DataFrame containing the number of notes played in each interval.
+
+    Parameters
+    ----------
+    record : pd.DataFrame
+        DataFrame contining data about the notes played
+    duration : int, optional
+        The duration of the intervals in which we want to count the notes, by default 1
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame containing the number of notes played in each interval
+    """
+    intervals = get_intervals(record, duration)
+    times = record[["start", "pitch"]].groupby("start").count()
+    nps = times.groupby(pd.cut(times.index, intervals)).sum() / duration
+    nps["count"] = nps["pitch"]
+    nps = nps.drop("pitch", axis=1)
+    return nps
+
+
+def count_notes_in_interval(record: pd.DataFrame, interval: pd.Interval) -> int:
+    """
+    Counts the number of notes played in a given interval.
+
+    Parameters
+    ----------
+    record : pd.DataFrame
+        DataFrame containing data about the notes played
+    interval : pd.Interval
+        Interval in question
+
+    Returns
+    -------
+    int
+        Number of notes played in the given interval
+    """
+    return record.loc[(record["start"] >= interval.left) & (record["end"] < interval.right)]["pitch"].count()
+
+
+def count_notes_by_intervals(record: pd.DataFrame, duration: int) -> pd.DataFrame:
+    """
+    For a given record and duration, counts the number of notes played in every interval of given duration that starts on a note.
+
+    Parameters
+    ----------
+    record : pd.DataFrame
+        DataFrame containing data about the notes played
+    duration : int
+        The duration of the intervals
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame whose index is a pd.Interval object with the column 'notes_played' contains the number of notes played in that interval.
+    """
+    intervals = get_intervals_by_note(record, duration)
+
+    rv = {interval: count_notes_in_interval(record, interval) for interval in intervals}
+    rv = pd.DataFrame(list(rv.items()), columns=["interval", "notes_played"]).set_index("interval")
+    return rv
+
+
+def fastest_interval(record: pd.DataFrame, duration: int) -> Tuple[pd.Interval, int]:
+    """
+    For a given record and interval duration, returns the interval which has the most notes played.
+
+    Parameters
+    ----------
+    record : pd.DataFrame
+        DataFrame containing data about notes played
+    duration : int
+        The duration of the interval
+
+    Returns
+    -------
+    Tuple[pd.Interval, int]
+        A tuple (interval, count) where:
+        * interval represents the interval in which the notes are played the fastest
+        * count represents the number of notes played in said interval
+    """
+    interval_speed = count_notes_by_intervals(record, duration)
+    rv = interval_speed[interval_speed.notes_played == interval_speed.notes_played.max()].iloc[0]
+    return record, rv.name, rv.notes_played
```

Comment:
What's the point of returning `record` here? It wasn't modified in any way, and also it doesn't match the return type hint, so I'm guessing it's tech-debt :)

Reply:
Yes, the parameter is returned unchanged. This is because the function is run in parallel so I need some way to recover the record that contains the fastest 15 seconds. The edit was made without altering the docs.

Reply:
This is an error, no point in returning the record.